extension/src/app.rs

```diff
@@ -0,0 +1,247 @@
+use crate::{
+    components::{CreateWallet, TradeInfo, UnlockWallet, WalletDetails},
+    event_bus::EventBus,
+    wallet_updater::WalletUpdater,
+};
+use js_sys::Promise;
+use message_types::bs_ps::{self, BackgroundStatus, TransactionData, WalletStatus};
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_extension::browser;
+use wasm_bindgen_futures::{spawn_local, JsFuture};
+use yew::prelude::*;
+
+// We do not support renaming the wallet for now
+pub const WALLET_NAME: &str = "demo-wallet";
+
+pub struct App {
+    link: ComponentLink<Self>,
+    state: State,
+    _event_bus: Box<dyn Bridge<EventBus>>,
+    _wallet_updater: WalletUpdater,
+}
+
+pub enum Msg {
+    CreateWallet,
+    UnlockWallet,
+    BackgroundStatus(Box<BackgroundStatus>),
+    SignAndSend { tx_hex: String, tab_id: u32 },
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct State {
+    wallet_name: String,
+    wallet_password: String,
+    wallet_status: WalletStatus,
+    sign_tx: Option<TransactionData>,
+}
+
+impl Component for App {
+    type Message = Msg;
+    type Properties = ();
+
+    fn create(_props: Self::Properties, link: ComponentLink<Self>) -> Self {
+        log::debug!("PopupApp: creating...");
+        let mut wallet_updater = WalletUpdater::new();
+        wallet_updater.spawn();
+
+        let callback = link.callback(Msg::BackgroundStatus);
+        App {
+            link,
+            state: State {
+                wallet_name: WALLET_NAME.to_string(),
+                wallet_password: "".to_string(),
+                wallet_status: WalletStatus::None,
+                sign_tx: None,
+            },
+            _event_bus: EventBus::bridge(callback),
+            _wallet_updater: wallet_updater,
+        }
+    }
+
+    fn update(&mut self, msg: Self::Message) -> ShouldRender {
+        match msg {
+            Msg::UnlockWallet => {
+                let inner_link = self.link.clone();
+                let msg = bs_ps::Message {
+                    rpc_data: bs_ps::RpcData::UnlockWallet(
+                        self.state.wallet_name.clone(),
+                        self.state.wallet_password.clone(),
+                    ),
+                    target: message_types::Component::Background,
+                    source: message_types::Component::PopUp,
+                    content_tab_id: 0,
+                };
+                send_to_backend(
+                    msg,
+                    Box::new(move |response| {
+                        if let Ok(response) = response {
+                            if let Ok(status) = response.into_serde() {
+                                inner_link.send_message(Msg::BackgroundStatus(status));
+                            }
+                        }
```

Comment:
TODO: Do not ignore these errors. Log and maybe display something on the extension.